packages/flutter_tools/lib/src/migrate/migrate_config.dart

```diff
@@ -0,0 +1,257 @@
+  static Future<List<MigrateConfig>> parseOrCreateMigrateConfigs({List<String>? platforms, Directory? projectDirectory, String? currentRevision, String? createRevision, bool create = true}) async {
+    platforms ??= getSupportedPlatforms(includeRoot: true, flutterProject: projectDirectory == null ? null : FlutterProject.fromDirectory(projectDirectory));
+    final List<MigrateConfig> configs = <MigrateConfig>[];
+    for (final String platform in platforms) {
+      if (MigrateConfig.getFileFromPlatform(platform, projectDirectory: projectDirectory).existsSync()) {
+        // Existing config. Parsing.
+        configs.add(MigrateConfig.fromFile(getFileFromPlatform(platform, projectDirectory: projectDirectory)));
+      } else {
+        // No config found, creating empty config.
+        final MigrateConfig newConfig = MigrateConfig(
+          platform: platform,
+          createRevision: createRevision,
+          baseRevision: currentRevision,
+          unmanagedFiles: _kDefaultUnmanagedFiles[platform] != null ? _kDefaultUnmanagedFiles[platform]! : <String>[],
+        );
+        if (create) {
+          newConfig.writeFile(projectDirectory: projectDirectory);
+        }
+        configs.add(newConfig);
+      }
+    }
+    return configs;
+  }
+
+  /// Returns the File that the migrate config belongs given a platform and a project directory.
+  static File getFileFromPlatform(String? platform, {Directory? projectDirectory}) {
+    Directory? platformDir;
+    final FlutterProject project = projectDirectory == null ? FlutterProject.current() : FlutterProject.fromDirectory(projectDirectory);
+    switch (platform) {
+      case 'root': {
+        platformDir = project.directory;
+        break;
+      }
+      case 'android': {
+        platformDir = project.android.hostAppGradleRoot;
+        break;
+      }
+      case 'ios': {
+        platformDir = project.ios.hostAppRoot;
+        break;
+      }
+      case 'web': {
+        platformDir = project.web.directory;
+        break;
+      }
+      case 'macos': {
+        platformDir = project.macos.hostAppRoot;
+        break;
+      }
+      case 'linux': {
+        platformDir = project.linux.managedDirectory.parent;
+        break;
+      }
+      case 'windows': {
+        platformDir = project.windows.managedDirectory.parent;
+        break;
+      }
+      case 'windowsUwp': {
+        platformDir = project.windowsUwp.managedDirectory.parent;
+        break;
+      }
+      case 'fuchsia': {
+        platformDir = project.fuchsia.editableHostAppDirectory;
+        break;
+      }
+    }
+    if (platformDir == null) {
+      throwToolExit('Invalid platform when creating MigrateConfig', exitCode: 1);
+    }
+    return platformDir.childFile(kFileName);
+  }
+
+  /// Returns a list of platform names that are supported by the project.
+  static List<String> getSupportedPlatforms({bool includeRoot = false, FlutterProject? flutterProject}) {
```

Comment:
this seems like a utility that belongs in FlutterProject. A method that provides all the supported platforms.

Reply:
Moved.